### PR TITLE
Feature/66/ms sql

### LINKED ISSUE #66: SQL Server Module
Are there plans or a timeline for a cloud sql module for sql server?
Thanks

### PR DESCRIPTION
**Pull request by kopachevsky**
* Fixes #66

### CODE REVIEW

File: modules/mssql/main.tf [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  count      = length(var.additional_databases)
+  project    = var.project_id
+  name       = var.additional_databases[count.index]["name"]
+  charset    = lookup(var.additional_databases[count.index], "charset", "")
+  collation  = lookup(var.additional_databases[count.index], "collation", "")
+  instance   = google_sql_database_instance.default.name
+  depends_on = [google_sql_database_instance.default]
+}
+
+resource "random_id" "user-password" {
```
Comment (paulpalamarchuk) [inline modules/mssql/main.tf]: Use resource `random_password` instead

File: test/fixtures/mssql-ha/main.tf [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+locals {
+  /*
+    Random instance name needed because:
+    "You cannot reuse an instance name for up to a week after you have deleted an instance."
+    See https://cloud.google.com/sql/docs/mysql/delete-instance for details.
+  */
+  instance_name = join("-", [var.ha_name, random_id.instance_name_suffix.hex])
+}
+
+module "mssql" {
```
Comment (paulpalamarchuk) [inline test/fixtures/mssql-ha/main.tf]: Use examples, please

File: test/fixtures/mssql-simple/main.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+locals {
+  /*
+    Random instance name needed because:
+    "You cannot reuse an instance name for up to a week after you have deleted an instance."
+    See https://cloud.google.com/sql/docs/mysql/delete-instance for details.
+  */
+  instance_name = join("-", [var.name, random_id.instance_name_suffix.hex])
+}
+
+module "mssql" {
```
Comment (paulpalamarchuk) [inline test/fixtures/mssql-simple/main.tf]: Use examples, please

File: modules/mssql/variables.tf [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+
+variable "tier" {
+  description = "The tier for the master instance."
+  type        = string
+  default     = "db-f1-micro"
+}
+
+variable "zone" {
+  type        = string
+  description = "The zone for the master instance, it should be something like: `a`, `c`."
```
Comment (paulpalamarchuk) [inline modules/mssql/variables.tf]: Not sure if it's good format. We are using full name of zone in other modules.

**Comment by paulpalamarchuk** [comment]
Code looks good. Waiting for tests.

File: test/fixtures/mssql-simple/main.tf (hunk 3 above)
Comment (kopachevsky) [reply test/fixtures/mssql-simple/main.tf]: where are no example for mssql ha yet

Comment (paulpalamarchuk) [reply test/fixtures/mssql-simple/main.tf]: Then it's ok

**Comment by sclausson** [comment]
@paulpalamarchuk and @kopachevsky are there plans to merge this PR to master? We're also wanting to use this module to create SQL instances. Thanks!

File: modules/mssql/main.tf [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    disabled = {}
+  }
+
+  user_labels_including_tf_dependency = {
+    enabled  = merge(map("tf_dependency", var.peering_completed), var.user_labels)
+    disabled = var.user_labels
+  }
+}
+
+
```
Comment (namusyaka) [inline modules/mssql/main.tf:33]: unnecessary whitespace ✂️

**Comment by morgante** [comment]
@namusyaka Would you mind actually taking this PR over? If you do and fix up the changes you found I can merge quickly.

**Comment by namusyaka** [comment]
@morgante no, I'll do it next week.

**Comment by namusyaka** [comment]
These commits were merged in #112 . Thanks for your contribution @kopachevsky!
